azure/converters/vm.go

```diff
@@ -47,19 +46,19 @@ type VM struct {
 }
 
 // SDKToVM converts an Azure SDK VirtualMachine to the CAPZ VM type.
-func SDKToVM(v compute.VirtualMachine) *VM {
+func SDKToVM(v armcompute.VirtualMachine) *VM {
 	vm := &VM{
 		ID:    ptr.Deref(v.ID, ""),
 		Name:  ptr.Deref(v.Name, ""),
-		State: infrav1.ProvisioningState(ptr.Deref(v.ProvisioningState, "")),
+		State: infrav1.ProvisioningState(ptr.Deref(v.Properties.ProvisioningState, "")),
 	}
 
-	if v.VirtualMachineProperties != nil && v.VirtualMachineProperties.HardwareProfile != nil {
-		vm.VMSize = string(v.VirtualMachineProperties.HardwareProfile.VMSize)
+	if v.Properties != nil && v.Properties.HardwareProfile != nil {
+		vm.VMSize = string(*v.Properties.HardwareProfile.VMSize)
 	}
 
-	if v.Zones != nil && len(*v.Zones) > 0 {
-		vm.AvailabilityZone = azure.StringSlice(v.Zones)[0]
+	if len(v.Zones) > 0 {
+		vm.AvailabilityZone = *v.Zones[0]
```

Comment:
I suppose depending on your level of paranoia, we could make sure this isn't nil too.